### DISCUSSION TITLE
Desktop: Takeout upload very slow

### DISCUSSION THREAD

**Discussion by philips** [ente-io/ente#2145] [General]
### Description

I really want to use and support Ente. But, I have about 1TB of data in Google Photos. And uploading 1 of 20 of my 50GB Google Takeout archives takes hours and hours (I have never completed an upload in the last 3 days) on my M2 Macbook (overnight for 8hrs only partially uploaded 1 50GB upload and I had to cancel it to get to work). A single 50GB download from Google took about 1hr.

I have 1Gb/s symmetric fiber internet at home. 

### Version

1.7.0

### What product are you using?

Ente Photos

### What platform are you using?

Desktop - macOS

**Comment by mnvr** [comment]
The use of ALL CAPS ASSERTIONS is not appreciated.

Google photos is not end to end encrypted, and can do server side optimizations and pre-computations that Ente cannot.

There are things we can improve. For example, we currently do 4 parallel uploads. The number 4 is a rough heuristic - on slower machines the parallel encryption of more than 4 streams overloads the host machines. But for people with newer machines, 4 might be too conservative.

If you have other concrete suggestions, do mention.

**Comment by philips** [comment]
All caps is to express frustration at a commercial product advertising that it support Google Takeout but also being so slow as to be useless. To actually move my account to the ente plan, that I am now $20 deep into funding, would take weeks and I can't trust it is actually going to complete given how often the ente desktop app seems to just blank out with a black box. I will once again cancel using ente for now. I am sorry you don't appreciate the sentiment.

I understand ente is end to end encryption and that it can be computationally complicated. But, I can't imagine how _partially_ encrypting 50GB of images can take well over 8 hours on a top of the line laptop. This feature is essentially useless at this speed and you should rethink advertising it until it can be sped up.

**Comment by philips** [comment]
Maybe you can reduce user frustration by benchmarking and estimating how many megabytes a minute your software can handle. I did read the support docs before purchasing and the speed nor computational cost is documented. And I would argue it is extremely material information to disclose given my own experience here. https://ente.io/faq/migration/from-google-photos/

**Comment by mnvr** [comment]
Tracking this task at https://github.com/ente-io/ente/discussions/3420
